### PR TITLE
Add instance: Hashable SomeLspId

### PR DESCRIPTION
**Pull request by thomasjm**
Just an instance I've had to include separately in projects in the past.

### CODE REVIEW

File: lsp-types/src/Language/LSP/Types/LspId.hs [hunk 1]
```diff
@@ -45,3 +46,6 @@ instance Ord SomeLspId where
       go (IdString a) (IdString b) = a `compare` b
       go (IdInt    _) (IdString _) = LT
       go (IdString _) (IdInt    _) = GT
+
+instance Hashable SomeLspId where
+  hashWithSalt n (SomeLspId a) = hashWithSalt n (toBase a)
```
Comment (michaelpj) [inline lsp-types/src/Language/LSP/Types/LspId.hs:51]: I'm a little confused. Why not derive `Hashable` for `LspId` and then just define this in terms of that?

Comment (thomasjm) [reply lsp-types/src/Language/LSP/Types/LspId.hs:51]: Oh sure, that works too

Comment (thomasjm) [reply lsp-types/src/Language/LSP/Types/LspId.hs:51]: I have no idea why it is like this actually, this is old code haha. Sending that change now...

**Review by michaelpj — APPROVED** [review]
Lovely, thanks.

**Pull request merged**